pypsrp/wsman.py

```diff
@@ -773,6 +783,21 @@ def _build_session(self):
         elif self.no_proxy:
             session.proxies = orig_proxy
 
+        # Retry on connection errors, with a backoff factor
+        retries = requests.packages.urllib3.util.retry.Retry(
+            total=self.reconnection_retries,
+            connect=self.reconnection_retries,
+            read=self.reconnection_retries,
+            backoff_factor=self.reconnection_backoff,
+            status_forcelist=[425, 429, 503],
```

Comment:
Why these status codes?

Reply:
All three of these are HTTP status codes that allow for retrying.

Reply:
In fact HTTP 413, HTTP 429 and HTTP 503 return a Retry-After header that influences the backoff value. This way the server can tell the client to come back at some later interval (because it hasn't finished starting, or is handling too many connections).

Before our implementation would ignore those. (TBH I don't know if Microsoft has any of this implemented, but it doesn't hurt to do it anyway because a proxy or gateway could be used)